src/watch.ts

```diff
@@ -47,6 +52,14 @@ export class Watch {
    *      until an other package is updated.
    *      It will never be up to date because he receive event at the same pace
    *      as they arrive in listener A, even if it's not the same package.
+   *
+   * --- Retry strategy.
+   *  Each packages update needs to be processed in order so there is a first naive strategy that will block the queue.
+   *  It will retry the same package after an exponential backoff, N times.
+   *
+   *  After N times, this update will be discarded in the this.skipped
+   *  This Map will be regularly reprocessed to avoid loosing jobs.
+   *  This is an in-memory only retry, if the process is stopped, skipped job are lost.
```

Comment:
maybe this should be persisted somewhere? Index settings? Or would you rely on bootstrap (very infrequent) for that

Reply:
Yeah, probably the next step. Two solutions: 
- a persistent DB but that would require more infra, making this project a bit more complexe to manage. eg: redis in dev vs redis in prod, means docker-compose + a redis instance in GCP + monitoring etc...
- using a dedicated algolia index. While algolia is not perfect for queuing, it is a simple and effective solution if we don't perform too much operation on it.

Reply:
I'm ok with going for the simpler (algolia) solution, as it doesn't require provisioning new environments and stays easy to run locally